lib/stat_tracker.rb

```diff
@@ -166,6 +166,91 @@ def total_games_by_team
     games_by_team
   end
 
+  def count_of_teams
+    @team_collection.teams.length
+  end
+
+  def worst_defense
+    # Name of the team with the highest average number of goals
+    # allowed per game across all seasons.
+    #find all games per team
+    teams_per_game = @gtc.game_teams.reduce(Hash.new(nil)) do |games_played, game_team|
+      if games_played[game_team.game_id] == nil
+        games_played[game_team.game_id] = [game_team]
+      elsif games_played[game_team.game_id] != nil
+        games_played[game_team.game_id] << game_team
+      end
+      games_played
+    end
+    # find other teams stats from that game
+    require "pry"; binding.pry
+    
+
+    #points-allowed-per-game by team
+  end
+
+  def best_fans
```

Comment:
This method is incredibly large!  Fortunately we can break it apart into lots of helpers or integrate the methods I wrote that do most of the same thing.  Great chance to refactor!